measure_test.go

```diff
@@ -0,0 +1,160 @@
+package measure
+
+import (
+	"testing"
+
+	"github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-datastore/query"
+	"github.com/ipfs/go-metrics-interface"
+)
+
+var (
+	repo map[string]float64 = make(map[string]float64)
+)
+
+type metricsMock string
+
+func (m metricsMock) Counter() metrics.Counter { return m }
+func (m metricsMock) Gauge() metrics.Gauge { return m }
+func (m metricsMock) Histogram(_ []float64) metrics.Histogram { return m }
+func (m metricsMock) Summary(_ metrics.SummaryOpts) metrics.Summary { return m }
+
+func (m metricsMock) Set(x float64) { repo[string(m)] = x }
+func (m metricsMock) Add(x float64) { repo[string(m)] += x }
+func (m metricsMock) Sub(x float64) { repo[string(m)] -= x }
+func (m metricsMock) Inc() { repo[string(m)] += 1 }
+func (m metricsMock) Dec() { repo[string(m)] -= 1 }
+func (m metricsMock) Observe(x float64) { repo[string(m)] = x }
+
+func newMockCreator(name, _ string) metrics.Creator {
+	return metricsMock(name)
+}
+
+func TestPutCallsGetMeasured(t *testing.T) {
+	metrics.InjectImpl(newMockCreator)
+	defer metrics.InjectImpl(nil)
```

Comment:
This won't work as it won't allow you inject new implementation when one is already injected.

I might have to think how to resolve it.

Allowing unregistering the injected creator might not be a bad idea.